### PR TITLE
Adds support for UICollectionViewCells without a nib/xib

### PR DESCRIPTION
**Pull request by swoolcock**
Closes issue #6 

Also fixes a crashing test due to unwrapping nil optionals.

### CODE REVIEW

**Comment by gwennguihal** [comment]
Thank you for the pull request!

File: Collor/Classes/CollectionDataSource.swift [hunk 1]
```diff
@@ -37,8 +37,21 @@ public class CollectionDataSource: NSObject, UICollectionViewDataSource {
...
-            let nib = UINib(nibName: cellDescriptor.className, bundle: cellDescriptor.bundle)
-            collectionView.register(nib, forCellWithReuseIdentifier: cellDescriptor.identifier)
+            if cellDescriptor.bundle.path(forResource: cellDescriptor.className, ofType: "nib") != nil {
+                let nib = UINib(nibName: cellDescriptor.className, bundle: cellDescriptor.bundle)
+                collectionView.register(nib, forCellWithReuseIdentifier: cellDescriptor.identifier)
+            }
+            else if let cellClass = Bundle.main.classNamed(cellDescriptor.className) as? UICollectionViewCell.Type {
+                collectionView.register(cellClass, forCellWithReuseIdentifier: cellDescriptor.identifier)
+            }
+            else if let bundleName = Bundle.main.object(forInfoDictionaryKey: "CFBundleName") as? String,
```
[the remark this answers is not in the document]
Comment (swoolcock) [reply Collor/Classes/CollectionDataSource.swift]: The first condition attempts to find the class from the fully qualified class name, which would include the module name.  This is what you get from `NSStringForClass`, such as `MyModule.MyClass`.

The second condition takes the name of the main bundle and attempts to use that as the module name, such that you could pass in just `MyClass`.

The main bundle name is often, but not always the same as the module name.  In our case, the bundle name is different for alpha builds of the app, so we have to set the cell descriptor identifier as the fully qualified class name.
